logformat.md

```diff
@@ -0,0 +1,51 @@
+# Gazebo Log Format Update
+
+## Overview
+
+The Gazebo log format is an SQLite3 database that stores serialized protobuf
+messages along with topic names in a time sequence. Importantly, simulation
+state information is recorded as serialized Components. This relies on
+a unique type id to associate the data to a component type. This type
+information is available via c++ typeid, which is not readily
+cross-language. For example, a javascript logfile reader would not know how
+to associate a typeid with the correct component.
+
+
+State information encoded as components requires a log file reader to
+implement or use Gazebo’s ECS. This represents a large barrier to entry.
+Ideally, a user could query data from a log file without implementing
+a complete ECS. 
+
+## Requirements
+
+1. Cross-platform and cross-language
+    1. Logged data
+        1. The current log file format is dependent on C++ typeid when it stores serialized ECS Component data. Typeid information is not available in other languages, such as javascript.
+        1. It’s reasonable to assume that a language can use Protobuf messages.
+    1. Log file format
+        1. The mechanism used to record data should also be cross platform.
+1. Ease of use
+    1. Queryable
+        1. Support random access to logged data
+    1. Low barrier to entry
+        1. A user of the current log format needs to re-implement the Gazebo ECS.
+1. Efficient
```

Comment:
Support being "interrupted" in that the log file won't be corrputed/destroyed in the case that Gazebo is shut down incorrectly or has a segmentation fault.

Since user systems run directly in the executable, there is always a chance of segfaults and we should do as much as possible to preserve already written data in case that happens.